### ISSUE TITLE
Cluster_encoder with CLUSTER_INDEX_LABELING assigns same label for cluster and noise

### ISSUE THREAD

**Issue by Baschdl** [annoviko/pyclustering#596]
I have one big cluster with 95 elements and 5 noise data points in my data. When converting DBSCAN's `CLUSTER_INDEX_LIST_SEPARATION` encoding to `CLUSTER_INDEX_LABELING`, all data points (the ones in the cluster and the noise) are assigned the same cluster index.

``` python
print(len(data))
## 100
dbscan_instance = dbscan(np.array(data), eps=1.15e-8, neighbors=10)
dbscan_instance.process()
    
clusters = dbscan_instance.get_clusters()
noise = dbscan_instance.get_noise()
print(len(clusters[0]))
## 95
print(len(noise))
## 5

encoder = cluster_encoder(dbscan_instance.get_cluster_encoding(), 
                          dbscan_instance.get_clusters(), 
                          np.array(data));

encoder.set_encoding(type_encoding.CLUSTER_INDEX_LABELING);

clusters = encoder.get_clusters()
print(len(clusters))
## 100
print(clusters)
## [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
```

[data.npy.zip](https://github.com/annoviko/pyclustering/files/4519453/data.npy.zip)

**Comment by annoviko** [comment]
Hi @Baschdl ,

Thank you for the issue reporting.

**Comment by annoviko** [comment]
The issue is corrected. Now outliers are going to be marked by `NaN` values in case of labeling. The correction is available on the `0.9.dev` branch. And will be available in the next release of the library.
